Add page cache hit metric into query profile report

MishaDemianenko opened the pull request:
Using newly introduced page cache hit metric it should be much more easier
to see how particular query use data that are already in page cache.
Each page cache `hit` indicate that data from cache was used and there was no page fault
when particular price of data request was served.
Now when hit metric added into query execution profile statistic we have possibility to
investigate how particular query utilize page cache.

MishaDemianenko committed:
Add page cache hit metric into query profile report

Using newly introduced page cache hit metric it should be much more easier
to see how particular query use data that are already in page cache.
Each page cache `hit` indicate that data from cache was used and there was no page fault
when particular price of data request was served.
Now when hit metric added into query execution profile statistic we have possibility to
investigate how particular query utilize page cache.

systay commented on community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/spi/QueryContext.scala:
```diff
@@ -21,12 +21,12 @@ package org.neo4j.cypher.internal.compiler.v3_2.spi
 import org.neo4j.cypher.internal.frontend.v3_2.SemanticDirection
 import org.neo4j.graphdb.{Node, Path, PropertyContainer, Relationship}
-import org.neo4j.cypher.internal.compiler.v3_2.IndexDescriptor
+import org.neo4j.io.pagecache.tracing.cursor.PageCursorTracer
```
We should not introduce any new store dependencies here. Instead, create a trait in the compiler module, and implement it in the compatibility package. There you can have your kernel dependencies.

systay commented on community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_2/TransactionBoundQueryContext.scala:
```diff
@@ -745,6 +746,10 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
     transactionalContext.statement.schemaWriteOperations()
+
+  override def pageCursorTracer(): PageCursorTracer = {
+    transactionalContext.statement.asInstanceOf[KernelStatement].getPageCursorTracer
```
Shouldn't `getPageCursorTracer` be in the `Statement` interface?

systay reviewed (changes requested):
Looks good, except for the introduction of a kernel dependency in the SPI

MishaDemianenko committed:
Add page cache misses into profile information.

By adding page cache misses to a profile we allow user to have a full picture of hits and misses that
was observed during particular query execution.
That should provide additional insight and reasons for query execution timings.

MishaDemianenko committed:
Introduce KernelStatisticProvider to separate cypher from new store dependencies.
And ProfileKernelStatisticProvider as it de facto implementation that will
provide execution statistic.

MishaDemianenko committed:
Introduce ExecutionStatisticsOperations interface for kernel execution statistics exposure from statement.

MishaDemianenko commented:
@systay updated!

thobe commented:
In `CompositeIndexAcceptanceTest`: _Cannot get single property of multi-property index_.

Based on a bad base revision?

MishaDemianenko commented:
@thobe yep, since i will need to rebase anyway will wait for green build to do that :)

MishaDemianenko committed:
Code style fix.

MishaDemianenko commented:
@thobe rebased
